File: backend/src/main/java/com/dev/backend/service/CityService.java
package com.dev.backend.service;

import java.util.Date;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.dev.backend.enitity.City;
import com.dev.backend.repository.CityRepository;

@Service
public class CityService {
    
    @Autowired
    private CityRepository cityRepository;

    public List<City> searchAll(){
        return cityRepository.findAll();
    }

    public City insert(City city){
        city.setDateCreation(new Date());
        City cityNew = cityRepository.saveAndFlush(city);
        return cityNew;
    }

    public City alter(City city){
        city.setDateUpdate(new Date());
        return cityRepository.saveAndFlush(city);
    }

    public void delete(Long id){
        City city = cityRepository.findById(id).get();
        cityRepository.delete(city);
    }
}

File: frontend/src/service/register/CityService.js
import axios from 'axios';

export class CityService{

    url = process.env.REACT_APP_URL_API;

    city(){
        return axios.get(this.url+'/city/');
    }

    insert(city){
        return axios.post(this.url+'/city/', city);
    }

    alter(city){
        return axios.put(this.url+'/city/', city);
    }

    delete(id){
        return axios.delete(this.url+'/city/'+id);
    }
}